src/attributes/val.js

```diff
@@ -4,7 +4,8 @@ define( [
 	"../core/init"
 ], function( jQuery, support ) {
 
-var rreturn = /\r/g;
+var rreturn = /\r/g,
+	rspaces = /\s/g;
```

Comment:
[HTML whitespace](https://html.spec.whatwg.org/#space-character) is a smaller set than that matched by `\s`, and this doesn't account for longer runs. I think you want `/[\x20\t\r\n\f]+/g`, matching various other regular expressions in our codebase.